### ISSUE TITLE
Bug: Nodes flickering.

### ISSUE THREAD

**Issue by Gh0stBlade** [Nelarius/imnodes#46]
I'm encountering an issue where nodes flicker.

![Engine-2020-07-16-19-43-46-31](https://user-images.githubusercontent.com/6951235/87709889-df18ee00-c79c-11ea-9782-19f04a86a9e3.gif)

**Comment by Nelarius** [comment]
Hi @Gh0stBlade !

Not yet sure what could be causing this -- is there something specific which triggers the flickering (like after a node is created)? 🤔 Looks like the links flicker along with the node itself, so none of the pins or links are getting rendered for a moment either... Is it possible to provide a more minimal repro somehow?

**Comment by Gh0stBlade** [comment]
It's like something internally is not being cleared or I did something wrong.

I changed in my code:

```C
MaterialTextureNode::MaterialTextureNode(ID3D11ShaderResourceView* pTextureSRV)
{
    this->m_id = /*g_currentMaterialNodeIndex*/rand();
    this->pTextureSRV = pTextureSRV;
    g_currentMaterialNodeIndex += MaterialTextureNode::Attribute::COUNT;
    this->m_index = textureNodeCount++;
}
```
Rather than using hardcoded Id which worked for all other nodes I'm having to use rand. However, there is no issue with assigning unique hardcoded ids (it works for all other nodes) and i debugged to verify that all the nodes have their own unique id and they do.

Cheers.

**Comment by Nelarius** [comment]
And just to be clear, `g_currentMaterialNodeIndex` is used for all the different node types? Meaning that all nodes get distinct ids? In that case, `m_id` should be fine and there's probably something buggy in the library itself.

**Comment by Gh0stBlade** [comment]
Yes its used globally for all material node types. Everything 100% has its own id i checked nultiple times but the nodes flicker out of control.

**Comment by Nelarius** [comment]
Hi @Gh0stBlade ! Not sure if relevant anymore, but an issue related to nodes erroneously trading places every other frame was found and fixed in #58 . This may have been the same issue which you were facing. The bug only triggered when using a node id of zero.

**Comment by Gh0stBlade** [comment]
Issue is likely fixed now.
